src/routing/middleware/rate_limiter.rs

```diff
@@ -159,27 +159,49 @@ mod tests {
 
         ERROR_CONTEXT
             .scope(RequestType::Unknown, async {
-                let total_requests = 20;
-                let delay = Duration::from_millis((1000.0 / rps as f64) as u64);
-
-                // All requests submitted at the max rps rate should succeed.
-                for _ in 0..total_requests {
-                    sleep(delay).await;
+                // Test cases: (delay_ms, expected_status)
+                let delay_for_token = 110; // when a token should become available ~ 1000ms/rps + delta=100+10=110ms
```

Comment:
Hmm ideally we should avoid sleeping in the tests, but in this case it seems that we can't easily plug https://docs.rs/governor/0.10.0/governor/clock/struct.FakeRelativeClock.html in there

Reply:
Can you maybe try to use the https://docs.rs/tokio/latest/tokio/time/fn.advance.html instead of sleep?

Reply:
Governor uses [QuantaClock](https://paritytech.github.io/try-runtime-cli/governor/clock/struct.QuantaClock.html), which seems to not be affected by `advance()`. We could use a custom (mock) clock in testing, e.g. [FakeRelativeClock](https://paritytech.github.io/try-runtime-cli/governor/clock/struct.FakeRelativeClock.html). But in this case we should restructure code and create a Governor Layer with [RateLimiter::direct_with_clock(..)](https://paritytech.github.io/try-runtime-cli/governor/struct.RateLimiter.html#method.direct_with_clock). In this case prod and testing code would differ, which is not great.

Reply:
I think it is a good idea to avoid using `sleep()`, but in our case it would save maybe 0.5-1 sec

Reply:
Tower governor sadly doesn't support using custom clocks, I made a PR for them to add this but not sure if/when it will get merged. So let's skip for now.